### PR TITLE
Metrics improvements

### PR DESCRIPTION
**Pull request by levity**
## Description

- Adds a new `pageview` event in Mixpanel that fires each time react-router receives a new location (on initial load and during in-app navigation).
- Adds front-end Mixpanel SDK, through a proxy (at a new API endpoint `/api/mp/track`), for tracking pageviews when the user has not yet connected a wallet.
- Calls `identify` after successful login to link the Mixpanel-generated distinct ID with a profile ID in our system.
- Enabled when `REACT_APP_MIXPANEL_TOKEN` is set; calls to `track` and `identify` fail silently otherwise.
  - Also needs `REACT_APP_MIXPANEL_HOST` to be set for proxying. This is set to the staging app for all preview environments in Vercel.
  - Create `src/features/analytics` to group this code (used in both frontend & backend)

## Todo
- [x] create new API endpoint for tracking page views
- [x] create interaction events from page views (or send directly to Mixpanel, TBD)
- [x] fail silently when mixpanel token isn't set in env
- [x] test in preview -- console: `testMp()`
- [x] call identify in `useFinishAuth`
- [x] track pageviews, at least for invite links, before login
- [x] fix stack overflow in routes.test

### CODE REVIEW

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2102** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2102" target="_blank">Graphite</a> 
    * **PR #2119** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2119" target="_blank">Graphite</a>  👈
    * **PR #2106** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2106" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/2119?utm_source=stack-comment).

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2102** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2102" target="_blank">Graphite</a> 
    * **PR #2119** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2119" target="_blank">Graphite</a>  👈
    * **PR #2106** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2106" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/2119?utm_source=stack-comment).

File: api/log.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    );
+    assert(profile, 'Invalid authorization token');
+    await addPropsAndTrack({
+      profile_id: profile.id,
+      event_type: 'pageview',
+      created_at: new Date(),
+      id: 0,
+      data: {
+        path: normalizePath(location.pathname),
+        original_path: location.pathname,
```
Comment (CryptoGraffe) [inline api/log.ts:45]: in a future version it might be nice to have the referring path

**Review by CryptoGraffe — APPROVED** [review]
nice work

**Pull request merged**